hermes/probe/create/create.go

```diff
@@ -0,0 +1,186 @@
+)
+
+const (
+	FileNamePrefixSize      = 9    // len("Hermes_ID")
+	begin                   = 1    // begin is the ID of the first HermesFile.
+	end                     = 50   // end is the ID of the last HermesFile.
+	maxFileSizeBytes        = 1000 // maxFileSizeBytes is the maximum allowed file size in bytes.
+	hermesAPILatencySeconds = "hermes_api_latency_s"
+)
+
+type randomFile struct {
+	id   int32 // id is a positive integer in the range [begin,end]
+	size int   // size is the file size in bytes.
+}
+
+type randomFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+// Read implements the standard Read interface, it reads data from a randomFileReader.
+// Argument:
+//      buf: a byte slice serves as a buffer
+// Returns:
+//	n:  the size of the buffer buf following a Read operation
+//	err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
+func (r *randomFileReader) Read(buf []byte) (n int, err error) {
+	// check whether Read is done
+	if r.i >= r.size {
+		return 0, io.EOF
+	}
+	b := buf
+	if len(buf) > (r.size - r.i) {
+		// if the length of buffer is greater than the number of bytes left to read make the size of the buffer match the number of bytes left to read
+		b = buf[:(r.size - r.i)]
+	}
+	n, err = r.rand.Read(b) // n is now the length  of the buffer
+	if err != nil {
+		return n, err // in this case n = 0
+	}
+	r.i += n
+	return n, err
+}
+
+// NewReader creates a new randomFileReader and returns a pointer to it
+// Returns:
+//	*randomFileReader: a Reader that supports randomFiles. This Reader contains information about the file size in bytes as well as a pseudo-random bytes generator
+// Warning: The randomFileReader produced by NewReader is not thread safe.
+func (f *randomFile) NewReader() *randomFileReader {
+	//  id will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+	return &randomFileReader{
+		size: f.size,
+		rand: rand.New(rand.NewSource(int64(f.id))), // rand.NewSource() requires an int64 as an argument
+	}
+}
+
+func newRandomFile(fileID int32, fileSize int) (*randomFile, error) {
+	if fileID < begin || fileID > end {
+		return nil, fmt.Errorf("the file id provided %d wasn't in the required range [%d, %d]", fileID, begin, end)
+	}
+	if fileSize > maxFileSizeBytes || fileSize <= 0 {
+		return nil, fmt.Errorf("invalid argument: randomFile.size = %d; want 0 < size <= %d", fileSize, maxFileSizeBytes)
+	}
+	return &randomFile{id: fileID, size: fileSize}, nil
+}
+
+func (f *randomFile) checksum() ([]byte, error) {
+	r := f.NewReader()
+	h := sha1.New()
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("io.Copy: %v", err)
+	}
+	return h.Sum(nil), nil
+}
+
+func (f *randomFile) fileName() (string, error) {
+	checksum, err := f.checksum()
+	if err != nil {
+		return "", fmt.Errorf("{%d, %d}.checksum = nil,  %v", f.id, f.size, err)
```

Comment:
How about

`fmt.Errorf("failed to generate file checksum: %v", err)`

Reply:
@cianbgoog suggested changing the error messages to match the go syntax does this not apply in this case?